datastore/postgres/filesbylayer.go

```diff
@@ -0,0 +1,114 @@
+package postgres
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"strconv"
+	"time"
+
+	"github.com/jackc/pgx/v4"
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/promauto"
+	"github.com/quay/claircore"
+	"github.com/quay/claircore/indexer"
+)
+
+var (
+	filesByLayerCounter = promauto.NewCounterVec(
+		prometheus.CounterOpts{
+			Namespace: "claircore",
+			Subsystem: "indexer",
+			Name:      "filesbylayer_total",
+			Help:      "The count of all queries issued in the FilesByLayer method",
+		},
+		[]string{"query"},
+	)
+
+	filesByLayerDuration = promauto.NewHistogramVec(
+		prometheus.HistogramOpts{
+			Namespace: "claircore",
+			Subsystem: "indexer",
+			Name:      "filesbylayer_duration_seconds",
+			Help:      "The duration of all queries issued in the FilesByLayer method",
+		},
+		[]string{"query"},
+	)
+)
+
+func (s *IndexerStore) FilesByLayer(ctx context.Context, hash claircore.Digest, scnrs indexer.VersionedScanners) ([]claircore.File, error) {
+	const (
+		selectScanner = `
+		SELECT id
+		FROM scanner
+		WHERE name = $1
+		  AND version = $2
+		  AND kind = $3;
+		`
+		query = `
+		SELECT file.id,
+			   file.path
+		FROM file_scanartifact
+				 LEFT JOIN file ON file_scanartifact.file_id = file.id
+				 JOIN layer ON layer.hash = $1
+		WHERE file_scanartifact.layer_id = layer.id
+		  AND file_scanartifact.scanner_id = ANY($2);
+		`
+	)
```

Comment:
I keep waffling on whether we should keep this style of sql statement. It's this or `embed`.

Reply:
As discussed this will be in a separate PR